pandas/core/array_algos/transforms.py

```diff
@@ -10,9 +10,8 @@
 def shift(values: np.ndarray, periods: int, axis: int, fill_value) -> np.ndarray:
     new_values = values
 
-    if periods == 0:
-        # TODO: should we copy here?
-        return new_values
+    if periods == 0 or values.size == 0:
```

Comment:
do we have a test on the copy itself? in the consumer of this code

Reply:
disabling the copy here doesnt break any tests, so looks like "no".  will add one